nntrainer/layers/common_properties.h

```diff
@@ -56,6 +56,106 @@ class Unit : public nntrainer::Property<unsigned int> {
   bool isValid(const unsigned int &v) const override { return v > 0; }
 };
 
+/**
+ * @brief RAII class to define the connection spec
+ *
+ */
+class ConnectionSpec {
+public:
+  static std::string NoneType;
+
+  /**
+   * @brief Construct a new Connection Spec object
+   */
+  ConnectionSpec() = default;
+  /**
+   * @brief Construct a new Connection Spec object
+   *
+   * @param layer_ids_ layer ids that will be an operand
+   * @param op_type_ operator type
+   */
+  ConnectionSpec(const std::vector<Name> &layer_ids_,
+                 const std::string &op_type_ = ConnectionSpec::NoneType);
+
+  /**
+   * @brief Construct a new Connection Spec object
+   *
+   * @param rhs rhs to copy
+   */
+  ConnectionSpec(const ConnectionSpec &rhs);
+
+  /**
+   * @brief Copy assignment operator
+   *
+   * @param rhs rhs to copy
+   * @return ConnectionSpec&
+   */
+  ConnectionSpec &operator=(const ConnectionSpec &rhs);
+
+  /**
+   * @brief Move Construct Connection Spec object
+   *
+   * @param rhs rhs to move
+   */
+  ConnectionSpec(ConnectionSpec &&rhs) noexcept;
+
+  /**
+   * @brief Move assign a connection spec operator
+   *
+   * @param rhs rhs to move
+   * @return ConnectionSpec&
+   */
+  ConnectionSpec &operator=(ConnectionSpec &&rhs) noexcept;
+
+  /**
+   * @brief Get the Op Type object
+   *
+   * @return const std::string& op_type (read-only)
+   */
+  const std::string &getOpType() const { return op_type; }
+
+  /**
+   * @brief Get the Layer Ids object
+   *
+   * @return const std::vector<Name>& vector of layer ids (read-only)
+   */
+  const std::vector<Name> &getLayerIds() const { return layer_ids; }
```

Comment:
can you also add a method to replace a given layer_id with another? this is required in realization methods of graph.

Reply:
To replace input_layers with this, there are number of helper methods need be added.
Let me wait for the reconstruction you are doing, because there are some inherent problem with current format. (eg) returning `std::vector<std::string> &` is impossible.)

Reply:
Let's merge this PR as #1206 is depending upon. Changes will be additive.

Reply:
cool.